pkg/ruler/base/notifier.go

```diff
@@ -88,10 +76,46 @@ func (rn *rulerNotifier) stop() {
 	rn.wg.Wait()
 }
 
+func getAlertmanagerTenantConfig(amConfig ruler_config.AlertManagerConfig, amOverrides ruler_config.AlertManagerConfig) (ruler_config.AlertManagerConfig, error) {
+	if amOverrides.AlertmanagerURL != "" {
+		amConfig.AlertmanagerURL = amOverrides.AlertmanagerURL
+	}
+
+	if len(amOverrides.AlertRelabelConfigs) > 0 {
+		amConfig.AlertRelabelConfigs = amOverrides.AlertRelabelConfigs
+	}
+
+	if amOverrides.AlertmanagerDiscovery {
+		amConfig.AlertmanagerDiscovery = amOverrides.AlertmanagerDiscovery
+	}
+
+	if amOverrides.AlertmanangerEnableV2API {
+		amConfig.AlertmanangerEnableV2API = amOverrides.AlertmanangerEnableV2API
+	}
+
+	if amOverrides.AlertmanagerRefreshInterval > 0 {
+		amConfig.AlertmanagerRefreshInterval = amOverrides.AlertmanagerRefreshInterval
+	}
+
+	if amOverrides.NotificationQueueCapacity > 0 {
+		amConfig.NotificationQueueCapacity = amOverrides.NotificationQueueCapacity
+	}
+
+	if amOverrides.NotificationTimeout > 0 {
+		amConfig.NotificationTimeout = amOverrides.NotificationTimeout
+	}
+
+	if err := mergo.Merge(&amConfig.Notifier, amOverrides.Notifier, mergo.WithOverride); err != nil {
+		return amConfig, fmt.Errorf("failed to apply alertmanager notifier limits config: %w", err)
+	}
+
+	return amConfig, nil
+}
+
 // Builds a Prometheus config.Config from a ruler.Config with just the required
 // options to configure notifications to Alertmanager.
-func buildNotifierConfig(rulerConfig *Config) (*config.Config, error) {
-	amURLs := strings.Split(rulerConfig.AlertmanagerURL, ",")
+func buildNotifierConfig(amConfig ruler_config.AlertManagerConfig, externalLabels labels.Labels) (*config.Config, error) {
```

Comment:
`ruler_config.AlertManagerConfig` is a big struct to be copying each time we call this function; can you use a pointer pls?

Reply:
On second thought, this function is called only a couple times so ignore my comment